irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -271,7 +271,19 @@ DROP TABLE IF EXISTS index_by_id_height_asset;
     }
 
     std::shared_ptr<WsvQuery> StorageImpl::getWsvQuery() const {
-      return wsv_;
+      auto postgres_connection =
+          std::make_unique<pqxx::lazyconnection>(postgres_options_);
+      try {
+        postgres_connection->activate();
+      } catch (const pqxx::broken_connection &e) {
+        exit(1);
```

Comment:
It is better to replace this with `throw` and TODO with task to change method signature.